### ISSUE TITLE
🐛 `has-[>svg]:px-2.5` not handled well by the tailwind parser

### ISSUE THREAD

**Issue by dyc3** [biomejs/biome#10368]
### Environment information

<details>

```bash
N/A

```

</details>

### What happened?

found this while testing the new tailwind rules

```
packages/@midpay/ui/src/lib/components/ui/input-group/input-group-button.svelte:10:65 parse ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ✖ Expected a value but instead found '[>svg]:p-0'.

     8 │ 			xs: "h-6 gap-1 rounded-[calc(var(--radius)-5px)] px-2 has-[>svg]:px-2 [&>svg:not([class*='size-'])]:size-3.5",
     9 │ 			sm: "h-8 gap-1.5 rounded-md px-2.5 has-[>svg]:px-2.5",
  > 10 │ 			"icon-xs": "size-6 rounded-[calc(var(--radius)-5px)] p-0 has-[>svg]:p-0",
       │ 			                                                             ^^^^^^^^^^
    11 │ 			"icon-sm": "size-8 p-0 has-[>svg]:p-0",
    12 │ 		},

     8 │ 			xs: "h-6 gap-1 rounded-[calc(var(--radius)-5px)] px-2 has-[>svg]:px-2 [&>svg:not([class*='size-'])]:size-3.5",
     9 │ 			sm: "h-8 gap-1.5 rounded-md px-2.5 has-[>svg]:px-2.5",
  > 10 │ 			"icon-xs": "size-6 rounded-[calc(var(--radius)-5px)] p-0 has-[>svg]:p-0",
       │ 			                                                             ^^^^^^^^^^
    11 │ 			"icon-sm": "size-8 p-0 has-[>svg]:p-0",
    12 │ 		},
```

the bug is that `has-[>svg]:px-2.5` is not being parsed correctly by the tailwind parser.

### Expected result

The `has-[>svg]` variant should have its value parsed as a css selector. it is currently being parsed as a value. There may be other variants that have this distinction.

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct
